File: packages/shared/src/health.ts
// SPDX-License-Identifier: AGPL-3.0-or-later
// Copyright (C) 2026 Salik Syed
/**
 * Health-check contract shared by both services.
 * Mirrors ./schemas/health.schema.json.
 */

export type HealthState = "ok" | "degraded";

export interface HealthResponse {
  /** Overall service state. */
  status: HealthState;
  /** Logical service name, e.g. "api" or "pipeline". */
  service: string;
  /** Service semantic version. */
  version: string;
  /** ISO-8601 timestamp the response was produced. */
  time: string;
}

File: services/pipeline/app/api/health.py
# SPDX-License-Identifier: AGPL-3.0-or-later
# Copyright (C) 2026 Salik Syed
"""Public health check — no auth, used by Docker/load balancers."""

from __future__ import annotations

from fastapi import APIRouter

from app import __version__
from app.api.schemas import HealthResponse

router = APIRouter()


@router.get("/health", response_model=HealthResponse, tags=["health"])
async def health() -> HealthResponse:
    return HealthResponse(service="pipeline", version=__version__)
